diff --git a/node_modules/react-native-screens/android/.gradle/8.9/checksums/checksums.lock b/node_modules/react-native-screens/android/.gradle/8.9/checksums/checksums.lock
new file mode 100644
index 0000000..e132841
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/8.9/checksums/checksums.lock differ
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/checksums/md5-checksums.bin b/node_modules/react-native-screens/android/.gradle/8.9/checksums/md5-checksums.bin
new file mode 100644
index 0000000..e1f5f77
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/8.9/checksums/md5-checksums.bin differ
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/checksums/sha1-checksums.bin b/node_modules/react-native-screens/android/.gradle/8.9/checksums/sha1-checksums.bin
new file mode 100644
index 0000000..fc8b165
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/8.9/checksums/sha1-checksums.bin differ
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/dependencies-accessors/gc.properties b/node_modules/react-native-screens/android/.gradle/8.9/dependencies-accessors/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/fileChanges/last-build.bin b/node_modules/react-native-screens/android/.gradle/8.9/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/8.9/fileChanges/last-build.bin differ
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/fileHashes/fileHashes.lock b/node_modules/react-native-screens/android/.gradle/8.9/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..c36f490
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/8.9/fileHashes/fileHashes.lock differ
diff --git a/node_modules/react-native-screens/android/.gradle/8.9/gc.properties b/node_modules/react-native-screens/android/.gradle/8.9/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..e8ebeb9
Binary files /dev/null and b/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/cache.properties b/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..7a6ab7e
--- /dev/null
+++ b/node_modules/react-native-screens/android/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Mon Mar 16 14:28:43 IST 2026
+gradle.version=8.9
diff --git a/node_modules/react-native-screens/android/.gradle/vcs-1/gc.properties b/node_modules/react-native-screens/android/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-screens/android/.project b/node_modules/react-native-screens/android/.project
new file mode 100644
index 0000000..1bfe1b5
--- /dev/null
+++ b/node_modules/react-native-screens/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-screens</name>
+	<comment>Project react-native-screens created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1772565172203</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-screens/android/build.gradle b/node_modules/react-native-screens/android/build.gradle
index 885faa6..30b84ba 100644
--- a/node_modules/react-native-screens/android/build.gradle
+++ b/node_modules/react-native-screens/android/build.gradle
@@ -94,7 +94,6 @@ if (isNewArchitectureEnabled()) {
     apply plugin: "com.facebook.react"
 }
 apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
 
 def reactNativeArchitectures() {
     def value = project.getProperties().get("reactNativeArchitectures")
@@ -199,6 +198,17 @@ android {
                 ]
             }
         }
+        kotlin {
+            if (IS_NEW_ARCHITECTURE_ENABLED) {
+                srcDirs += [
+                    "src/fabric/java",
+                ]
+            } else {
+                srcDirs += [
+                    "src/paper/java",
+                ]
+            }
+        }
         res {
             if (safeExtGet(['compileSdkVersion', 'compileSdk'], rnsDefaultCompileSdkVersion) >= 33) {
                 srcDirs = ["${androidResDir}/base", "${androidResDir}/v33"]
diff --git a/node_modules/react-native-screens/react-native.config.js b/node_modules/react-native-screens/react-native.config.js
index 7815576..f5e3cd4 100644
--- a/node_modules/react-native-screens/react-native.config.js
+++ b/node_modules/react-native-screens/react-native.config.js
@@ -1,24 +1,7 @@
 module.exports = {
   dependency: {
     platforms: {
 -      android: {
 -        componentDescriptors: [
 -          "RNSFullWindowOverlayComponentDescriptor",
 -          "RNSScreenContainerComponentDescriptor",
 -          "RNSScreenNavigationContainerComponentDescriptor",
 -          "RNSScreenStackHeaderConfigComponentDescriptor",
 -          "RNSScreenStackHeaderSubviewComponentDescriptor",
 -          "RNSScreenStackComponentDescriptor",
 -          "RNSSearchBarComponentDescriptor",
 -          'RNSScreenComponentDescriptor',
 -          "RNSScreenFooterComponentDescriptor",
 -          "RNSScreenContentWrapperComponentDescriptor",
 -          'RNSModalScreenComponentDescriptor',
 -          'RNSTabsHostComponentDescriptor',
 -          'RNSSafeAreaViewComponentDescriptor'
 -        ],
 -        cmakeListsPath: "../android/src/main/jni/CMakeLists.txt"
 -      },
 +      android: {},
     },
   },
 };
diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
index 0224a9e..2cbcf9a 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
@@ -423,9 +423,12 @@ class Screen(
         reactContext.runOnNativeModulesQueueThread(
             object : GuardedRunnable(reactContext.exceptionHandler) {
                 override fun runGuarded() {
-                    reactContext
-                        .getNativeModule(UIManagerModule::class.java)
-                        ?.updateNodeSize(id, width, height)
+                    val uiManager =
+                        runCatching {
+                            reactContext.getNativeModule(UIManagerModule::class.java)
+                        }.getOrNull()
+
+                    uiManager?.updateNodeSize(id, width, height)
                 }
             },
         )
diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
index b61c25a..265f361 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
@@ -191,7 +191,10 @@ class SafeAreaView(
                     edges = edges ?: SafeAreaViewEdges.ZERO,
                 )
             val reactContext = getReactContext(this)
-            val uiManager = reactContext.getNativeModule(UIManagerModule::class.java)
+            val uiManager =
+                runCatching {
+                    reactContext.getNativeModule(UIManagerModule::class.java)
+                }.getOrNull()
             if (uiManager != null) {
                 uiManager.setViewLocalData(id, localData)
                 // Sadly there doesn't seem to be a way to properly dirty a yoga node from java, so
diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreensShadowNode.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreensShadowNode.kt
index 9a7d10f..66d5cb0 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreensShadowNode.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreensShadowNode.kt
@@ -10,7 +10,11 @@ internal class ScreensShadowNode(
 ) : LayoutShadowNode() {
     override fun onBeforeLayout(nativeViewHierarchyOptimizer: NativeViewHierarchyOptimizer) {
         super.onBeforeLayout(nativeViewHierarchyOptimizer)
-        (context.getNativeModule(UIManagerModule::class.java))?.addUIBlock { nativeViewHierarchyManager: NativeViewHierarchyManager? ->
+        val uiManager =
+            runCatching {
+                context.getNativeModule(UIManagerModule::class.java)
+            }.getOrNull()
+        uiManager?.addUIBlock { nativeViewHierarchyManager: NativeViewHierarchyManager? ->
             if (nativeViewHierarchyManager == null) {
                 return@addUIBlock
             }
diff --git a/node_modules/react-native-screens/android/src/paper/java/com/swmansion/rnscreens/FabricEnabledHeaderConfigViewGroup.kt b/node_modules/react-native-screens/android/src/paper/java/com/swmansion/rnscreens/FabricEnabledHeaderConfigViewGroup.kt
index 7583109..f32af3d 100644
--- a/node_modules/react-native-screens/android/src/paper/java/com/swmansion/rnscreens/FabricEnabledHeaderConfigViewGroup.kt
+++ b/node_modules/react-native-screens/android/src/paper/java/com/swmansion/rnscreens/FabricEnabledHeaderConfigViewGroup.kt
@@ -45,6 +45,11 @@ abstract class FabricEnabledHeaderConfigViewGroup(
         lastHeight = height
 
         val reactContext = context as? ReactContext
-        val uiManagerModule = reactContext?.getNativeModule(UIManagerModule::class.java)
+        val uiManagerModule =
+            reactContext?.let {
+                runCatching {
+                    it.getNativeModule(UIManagerModule::class.java)
+                }.getOrNull()
+            }
         uiManagerModule?.setViewLocalData(this.id, PaddingBundle(height.toFloat(), paddingStart.toFloat(), paddingEnd.toFloat()))
     }
